src/logrotate.py

```diff
@@ -133,6 +134,7 @@ def _write_config(logrotate_config: LogrotateConfig) -> None:
         f"""{logrotate_config.log_path_glob_pattern} {{
 {logrotate_config.frequency}
 rotate {logrotate_config.rotate}
+copytruncate
```

Comment:
Please add this change (with an explanation of the necessity) to the PR description so we can track this change in the future more easily. 
This seems to fix the current situation for reactive logs too, so might also be worth adding to the changelog as it is a bug fix.